### ISSUE TITLE
Option to set metricbeat.config.modules.path

### ISSUE THREAD

**Issue by trenta** [corey-hammerton/puppet-metricbeat#14]
At the moment I can't find a way to set the metricbeat.config.modules.path option in metricbeat.yml and the recommended way to use modules now it so have the settings in the modules.d directory.  Adding this option would be handy.

**Comment by cdenneen** [comment]
So right now the modules array supplies the module config. Not sure why @corey-hammerton is creating modules.d files. It's one or other. @corey-hammerton How do you want this module to handle? By using @sfuerte hardcoded files it limits modifications. These should be templates.

**Comment by sfuerte** [comment]
@cdenneen, yes, templates is a proper way to do it. The current "files" implementation is just a quick and dirty fix to get them installed and enabled. Those files/modules come with standard Debian package. 
Simply didn't have time to do them in proper way via templates.

**Comment by cdenneen** [comment]
@sfuerte Thanks. Definitely works but not for any add/remove/updates going forward with new releases. Would you have the time to help implement templates for these?

**Comment by sfuerte** [comment]
@cdenneen, will see if I have time for that. 
On the another hand, had a conversation with sales guys from ES last week or so. Asked them a question why they don't provide official modules for Puppet to support file/metric/other beats. That would be very logical on their end to do that for most common orchestration systems (Puppet, Chef, Ansible, Terraform). Will ping them again, hopefully they can dedicate someone to do that.

**Comment by cdenneen** [comment]
Would be nice but believe it's due to limited "puppet" people at ES. @tylerjl is only 1 person ;)

**Comment by sfuerte** [comment]
I know, it's weird. I asked the sales guys about Puppet support, the answer was: "we do support Elastic there". Which is weird, given the proportion in environments and the worst comes to worst, ES itself can be installed manually. But for fully automated monitoring, one definitely doesn't want to install Beats manually.
